### PR TITLE
Enable Python 3.11 - 3.13 on Android

### PR DESCRIPTION
**Pull request by mhsmith**
All the necessary wheels have now been released.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by mhsmith** [comment]
iOS failed to get the test result twice in a row, on different Python versions, but that doesn't seem to have anything to do with this PR:
* [Python 3.9](https://github.com/beeware/Python-support-testbed/actions/runs/11459753675/job/31884993699)
* [Python 3.10](https://github.com/beeware/Python-support-testbed/actions/runs/11459436932/job/31883970921)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -106,22 +106,17 @@ jobs:
...
-          - backend: android
-            python-version: "3.11"
-          - backend: android
-            python-version: "3.12"
-          # Binary packages aren't available for 3.13 on Android yet
-          - backend: android
-            python-version: "3.13-dev"
+          # numpy and pandas aren't available for 3.13 on iOS
+          - backend: "iOS"
+            python-version: "3.13"
```
Comment (freakboy3742) [inline .github/workflows/ci.yml:111]: We can run the test suite on iOS; we just can't run the numpy and pandas tests; but they should be resilient to the package not being present.

**Review by freakboy3742 — COMMENTED** [review]
Android changes look good; but we can walk back the iOS changes. I'll get those rolled back now.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (freakboy3742) [reply .github/workflows/ci.yml]: For that matter, the 3.13 support packages exist for flatpaks and Windows.

**Review by freakboy3742 — APPROVED** [review]

**Pull request merged**